pyquil/api/_compiler.py

```diff
@@ -77,7 +78,8 @@ class QPUCompiler(AbstractCompiler):
     Client to communicate with the compiler and translation service.
     """
 
-    @_record_call
+    _calibration_program_lock: threading.Lock
```

Comment:
I wonder if this is the lock that can't be pickled in the QPU e2e tests...

Reply:
it makes sense - we should likely switch to `multiprocessing.Lock` to support that use case as well. There's probably a clever solution to use the right one for the runtime environment, but since runtime performance of this lock isn't important I think it's reasonable to use the more capable one

Reply:
Made the change.